### PR TITLE
config.py: Create empty data sub-directory

### PR DESCRIPTION
**Pull request by nvzard**
Remove need for the empty `data/` subdirectory.

Closes https://github.com/coala/corobo/issues/569

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -54,6 +54,16 @@
...
+
+def make_dir(path):
+    try:
+        os.mkdir(os.path.join(BOT_ROOT, path))
+    except OSError:
+        if os.path.isdir(os.path.join(BOT_ROOT, path)):
+            pass
+
+
+make_dir('data')
```
Comment (jayvdb) [inline config.py]: you should use `BOT_DATA_DIR`

File: config.py [hunk 2]
```diff
@@ -54,6 +54,16 @@
...
+
+def make_dir(path):
+    try:
+        os.mkdir(os.path.join(BOT_ROOT, path))
+    except OSError:
+        if os.path.isdir(os.path.join(BOT_ROOT, path)):
+            pass
+
+
+make_dir(BOT_DATA_DIR)
```
Comment (jayvdb) [inline config.py]: wtf.

This var is being set below

Comment (jayvdb) [reply config.py]: Also please add a note in the log that the directory is being created.

File: config.py [hunk 3]
```diff
@@ -54,7 +54,17 @@
 
 HIDE_RESTRICTED_COMMANDS = True
 
+
+def make_dir(path):
+    try:
+        os.mkdir(os.path.join(BOT_ROOT, path))
+    except OSError:
+        if os.path.isdir(os.path.join(BOT_ROOT, path)):
```
Comment (jayvdb) [inline config.py]: why not do this first before trying to make the directory?

will `os.mkdir` recursively create subdirectories that are missing?

Comment (nvzard) [reply config.py]: I removed the method completely and just did
```
if os.path.isdir(os.path.join(BOT_ROOT, BOT_DATA_DIR)):
    os.mkdir(os.path.join(BOT_ROOT, BOT_DATA_DIR))
```

**Comment by jayvdb** [comment]
ack https://github.com/coala/corobo/pull/591/commits/4c8bb8331b7869fc14d0f8a004a9020a177d52ef

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**